index.js

```diff
@@ -0,0 +1,140 @@
+    //open the dropdown
+    dropdownHeader.addEventListener('click', (event) => {
+        dropdownWrapper.classList.toggle('open');
+    });
+    //check if need to close the dropdown
+    document.addEventListener('click', (event) => {
+        if (dropdownWrapper.classList.contains('open') && !dropdownWrapper.contains(event.target)) {
+            dropdownWrapper.classList.remove('open');
+        }
+    });
+    //get the region that choosen
+    dropdownBody.addEventListener('click', (event) => {
+        const clickedElement = event.target;
+
+        if(clickedElement.tagName === 'LI')
+        {
+            selectedRegion= clickedElement.dataset.region;
+            console.log(selectedRegion);
+        }
+        // Re-render countries based on selected region
+        renderCountries(countries, container);
+    });
+  });
+
+
+function createCountryCard(country) {
+    // Create the main <a> element
+    const card = document.createElement('a');
+    card.className = 'country scale-effect';
+    card.href = '#';
+    card.dataset.countryName = country.name;
+  
+    // Create the flag container and image
+    const flagDiv = document.createElement('div');
+    flagDiv.className = 'country-flag';
+  
+    const flagImg = document.createElement('img');
+    flagImg.src = country.flag;
+    flagImg.alt = `${country.name} Flag`;
+  
+    flagDiv.appendChild(flagImg);
+  
+    // Create the info container
+    const infoDiv = document.createElement('div');
+    infoDiv.className = 'country-info';
+  
+    const countryTitle = document.createElement('h2');
+    countryTitle.className = 'country-title';
+    countryTitle.textContent = country.name;
+  
+    infoDiv.appendChild(countryTitle);
+  
+    // Create the ul
+    const ulList = document.createElement('ul');
+    ulList.className = 'country-brief';
+  
+    const populationItem = document.createElement('li');
+    populationItem.innerHTML = `<strong>Population: </strong>${country.population}`;
+  
+    const regionItem = document.createElement('li');
+    regionItem.innerHTML = `<strong>Region: </strong>${country.region}`;
+  
+    const capitalItem = document.createElement('li');
+    capitalItem.innerHTML = `<strong>Capital: </strong>${country.capital}`;
+  
+    ulList.appendChild(populationItem);
+    ulList.appendChild(regionItem);
+    ulList.appendChild(capitalItem);
+  
+    // Add the ul to the infoDiv
+    infoDiv.appendChild(ulList);
+  
+    // Add the flag and info containers to the main card
+    card.appendChild(flagDiv);
+    card.appendChild(infoDiv);
+    
+    // card.addEventListener('click',()=>{
+    //     location.href = "./details.html";
+    // });
+    card.addEventListener('click', () => {
+        const url = `details.html?name=${encodeURIComponent(country.name)}&region=${encodeURIComponent(country.region)}&population=${country.population}&capital=${encodeURIComponent(country.capital)}&flag=${encodeURIComponent(country.flag)}`;
```

Comment:
use a constant variable for the base url

```javascript
const baseUrl = 'details.html'
const url = `${baseUrl}?name={...}...`
```

moreover, you can create a new file `const.js` and define all your constants variable inside.